[Bug] PartitionExpire will fail when logOffsets is not empty.

### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.


### Paimon version

`partitionExpire` uses `tryOverwrite` for partition expiration, but when the `logOffsets` of `latestSnapshot` is not empty, it will cause an error. https://github.com/apache/incubator-paimon/blob/master/paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java#L622

```
        at org.apache.paimon.operation.FileStoreCommitImpl.tryCommitOnce(FileStoreCommitImpl.java:616) ~[ecom_paimon_udfs-1.0-SNAPSHOT.jar:?]
        at org.apache.paimon.operation.FileStoreCommitImpl.tryOverwrite(FileStoreCommitImpl.java:492) ~[ecom_paimon_udfs-1.0-SNAPSHOT.jar:?]
        at org.apache.paimon.operation.FileStoreCommitImpl.dropPartitions(FileStoreCommitImpl.java:370) ~[ecom_paimon_udfs-1.0-SNAPSHOT.jar:?]
        at org.apache.paimon.operation.PartitionExpire.doExpire(PartitionExpire.java:102) ~[ecom_paimon_udfs-1.0-SNAPSHOT.jar:?]
        at org.apache.paimon.operation.PartitionExpire.expire(PartitionExpire.java:85) ~[ecom_paimon_udfs-1.0-SNAPSHOT.jar:?]
        at org.apache.paimon.operation.PartitionExpire.expire(PartitionExpire.java:74) ~[ecom_paimon_udfs-1.0-SNAPSHOT.jar:?]
        at org.apache.paimon.table.sink.TableCommitImpl.expire(TableCommitImpl.java:212) ~[ecom_paimon_udfs-1.0-SNAPSHOT.jar:?]
        at org.apache.paimon.table.sink.TableCommitImpl.lambda$expire$2(TableCommitImpl.java:199) ~[ecom_paimon_udfs-1.0-SNAPSHOT.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: java.lang.UnsupportedOperationException
        at java.util.Collections$EmptyMap.putIfAbsent(Collections.java:4638) ~[?:?]
        at java.util.LinkedHashMap.forEach(LinkedHashMap.java:684) ~[?:?]
        at org.apache.paimon.operation.FileStoreCommitImpl.tryCommitOnce(FileStoreCommitImpl.java:559) ~[ecom_paimon_udfs-1.0-SNAPSHOT.jar:?]
        ... 10 more
```

### Compute Engine

flink-1.17

### Minimal reproduce step

1. set partition key for the table 
2. set expiration time for the partition
3. enable logSystem.

### What doesn't meet your expectations?

Partition cannot expire and be deleted.

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

--- a/paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java
+++ b/paimon-core/src/main/java/org/apache/paimon/operation/FileStoreCommitImpl.java
@@ -393,7 +393,7 @@ public class FileStoreCommitImpl implements FileStoreCommit {
                 Collections.emptyList(),
                 commitIdentifier,
                 null,
-                Collections.emptyMap());
+                new HashMap<>());
     }
 
     @Override

--- a/paimon-core/src/test/java/org/apache/paimon/operation/FileStoreCommitTest.java
+++ b/paimon-core/src/test/java/org/apache/paimon/operation/FileStoreCommitTest.java
@@ -609,7 +609,8 @@ public class FileStoreCommitTest {
         store.commitData(
                 data.values().stream().flatMap(Collection::stream).collect(Collectors.toList()),
                 gen::getPartition,
-                kv -> 0);
+                kv -> 0,
+                Collections.singletonMap(0, 1L));
 
         // generate partitions to be dropped
         ThreadLocalRandom random = ThreadLocalRandom.current();

